handbook/engineering/tuf.md

```diff
@@ -0,0 +1,138 @@
+
+## Add new TUF keys for authorized team members
+
+The CTO is responsible for determining who has access to push agent updates. Timestamp and Snapshot keys can be held online, so their use can be automated, but Targets and Root keys must always be held offline. The root keys are held by the CTO and CEO in secure locations. Root keys are retrieved once per year to rotate them before their annual expiration, or to sign for new Targets keys as needed. Targets keys may be generated to provide approved team members the ability to push agent updates to the TUF repo. 
+
+This process requires running TUF commands that are not available using the `fleetctl` binary, so the `tuf` CLI binary [documented by go-tuf](https://pkg.go.dev/github.com/theupdateframework/go-tuf#section-readme) needs to be downloaded and compiled for local use.
+
+There are two roles required to complete these steps, the "Root" role who holds the root keys, and the "Releaser" role, who is gaining access to push updates. 
+
+1. The Releaser creates a new local directory to store the TUF repo. The Releaser creates a sub-directory called `repository`.
+
+2. The Realeaser pulls down the contents of the TUF repo into the `repository` sub-directory. 
+
+3. From the root of their TUF directory, the Releaser run `tuf gen-key targets`. This will create a `keys` sub-directory and `staged` sub-directory. 
+
+4. The Releaser copies the `keys` directory to a USB drive, and deletes the `keys` directory from their local hard drive. 
+
+5. The Releaser send the `staged/root.json` to the Root role for signing. Note this file is safe to share and is publically available. 
+
+6. The Root role receives the `staged/root.json` file and copies it to a USB drive. 
+
+7. The Root role boots into the secure Ubuntu boot drive created during TUF repo creation. 
+
+8. The Root role connects the USB drive containing the `staged/root.json` file for signing. 
+
+9. The Root role connects the USB drive containing the root keys. 
+
+10. The Root role copies the `staged/root.json` onto the root keys USB at `staged/root.json`. 
+
+11. The root keys USB contains the `tuf` binary. Run `./tuf sign root.json` to sign the staged root metadata. 
+
+12. The Root role copies the signed `staged/root.json` back to the original USB drive they copied it from. 
+
+13. The Root role turns off the Ubuntu boot drive and accesses an online computer. 
+
+14. The Root role connects the USB drive containing the signed `staged/root.json` file and copies it to their local hard drive's TUF location in the same `staged/root.json`. 
+
+15. From the root of their local TUF repo, the Root role runs `tuf commit` to commit the staged root metadata to the `repository` directory. 
+
+16. The Root role pushes the updated contents of the `repository` directory to the remote TUF server. 
+
+17. The Releser role can now run `tuf sign` to sign agent updates using their offline Targets key.
+
+## Rotate the root keys 
```

Comment:
Am ok with using this new simpler approach, just checking. Also if we go this route, we might want to remove the `updates rotate root` functionality from `fleetctl`.

Reply:
I missed that we had `fleetctl updates rotate root`. That is simpler to use, but the problem I see is that we don't offer any way of adding more than one root key, which is advised as a best practice. That means the user would need to run `fleetctl updates rotate root`, then switch back to `tuf gen-key root`. It's the switching between CLIs mid-process that seems strange. What do you think?

I'm in favor of useing `fleetctl` but agree we should either extend it to enable using it for the full flow, or just use `tuf` since it has everything we need. I'm inclined to just use `tuf` now for root-based actions.

Reply:
+1 to using tuf for root rotation (the simpler, the better), and (later) let's remove the `updates rotate root` from `fleetctl`.